src/common/storage/backends/s3/Client.js

```diff
@@ -177,5 +174,14 @@ define([
         };
     }
 
+    S3Storage.prototype.stat = async function (path) {
+        const metadata = await this._stat(path, this.bucketName);
```

Comment:
It might be nice to remove `_stat` as it is only used in `stat` to simplify the code a little bit (the method name isn't terribly informative). Not a huge deal though